onboard-server/src/main/java/no/vegvesen/ixn/serviceprovider/OnboardRestController.java

```diff
@@ -289,13 +292,17 @@ public PrivateChannelApi addPrivateChannel(@PathVariable String serviceProviderN
 			throw new PrivateChannelException("Client channel cannot be null");
 		}
 
-		ServiceProvider serviceProviderToUpdate = getOrCreateServiceProvider(serviceProviderName);
-
-		PrivateChannel newPrivateChannel = serviceProviderToUpdate.addPrivateChannel(clientChannel.getPeerName());
+		boolean channelExists = privateChannelRepository.existsByPeerNameAndServiceProviderName(clientChannel.getPeerName(), serviceProviderName);
 
-		serviceProviderRepository.save(serviceProviderToUpdate);
-		OnboardMDCUtil.removeLogVariables();
-		return new PrivateChannelApi(newPrivateChannel.getPeerName(), newPrivateChannel.getQueueName(), newPrivateChannel.getId());
+		if(channelExists){
+			throw new PrivateChannelException("Client already has private channel");
+		}
+		else{
+			PrivateChannel newPrivateChannel = new PrivateChannel(clientChannel.getPeerName(), PrivateChannelStatus.REQUESTED, serviceProviderName);
+			privateChannelRepository.save(newPrivateChannel);
+			OnboardMDCUtil.removeLogVariables();
+			return new PrivateChannelApi(newPrivateChannel.getPeerName(), newPrivateChannel.getQueueName(), newPrivateChannel.getId(), newPrivateChannel.getServiceProviderName());
```

Comment:
Change the method signature to accept the new api object in orde to accept list of new private channels. Also change the return object and make new method in typeTransformer to return AddPrivateChannelsResponse